research-software-directory/RSD-as-a-service
# Software overview v2

# Software overview v2

Changes proposed in this pull request:
* Software overview with highlights, dynamic filters and view options for masonry, grid and list
* Page accessibility improvements based on Lighthouse audit suggestions
* Separate highlights page is removed
* Old software overview page is removed
* Unit tests created for new software overview page    
* Linting warning for use of img markup is turned off for the whole project
* BREAKING CHANGE comment is added to commit to bump major version number

How to test:
* `make start` to rebuild app
* login as rsd admin (professor3 if you have same .env values as me)
* navigate to admin - highlights section. Examine defined highlights. Add more or remove items.
* navigate to software (overview) page.
* confirm highlights are shown. When you remove all highlight items in the admin section, the highlight section will not be shown at all
* test order and filters on the left side of the page. After selection of an filter item the remaining filter options will be updated to remaining options present in the remaining selection. 
* test view options: masonry, grid or list. Change number of items per page.
* test pagination at the bottom of the page to navigate between the pages. Note! When using pagination the page will scroll to top.

## Example masonry
![image](https://github.com/research-software-directory/RSD-as-a-service/assets/9204081/e117f5b0-388d-4592-9ed0-c8a534588c8c)

## Example grid
![image](https://github.com/research-software-directory/RSD-as-a-service/assets/9204081/90afe59d-345f-4f0d-9178-772036d91454)

## Example list
![image](https://github.com/research-software-directory/RSD-as-a-service/assets/9204081/2e81df26-c0c8-4e98-8668-d5fe86f015c2)

PR Checklist:
